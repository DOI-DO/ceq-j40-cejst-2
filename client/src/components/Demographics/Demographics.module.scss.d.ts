declare namespace DemographicsNamespace {
    export interface IDemographicsScss {
      demographicsContainer: string;
      demographicItem: string;
    }
  }

declare const DemographicsScssModule: DemographicsNamespace.IDemographicsScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: DemographicsNamespace.IDemographicsScss;
  };

  export = DemographicsScssModule;
