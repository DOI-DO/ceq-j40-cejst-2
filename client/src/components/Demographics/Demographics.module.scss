@use '../../styles/design-system.scss' as *;

.demographicsContainer{
    @include u-padding(2);

    .demographicItem {
        display: flex;
        justify-content: space-between;
        
        span {
            font-size: .8rem;
            margin-top: 0;
        }
    }
}
