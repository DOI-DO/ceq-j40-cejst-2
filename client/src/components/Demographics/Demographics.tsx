import React from 'react';
import {Accordion} from '@trussworks/react-uswds';

import * as styles from './Demographics.module.scss';

// export interface IDemographicsProps {}

type IDemographicsData = {
  racial: [string, number][],
  age: [string, number][]
}

// Temporary while backend is developed
const demographicsData:IDemographicsData = {
  racial: [
    ['White', 61.4],
    ['Black or African Americon', 10.3],
    ['American Indian and Alaska Native', 10.3],
    ['American Indian and Alaska Native', 0.3],
    ['Asian', 7.2],
    ['Native Hawiian or Pacific Islander', 4.2],
    ['Other', 2.5],
    ['Two or more races', 6.7],
    ['Hispanic or Latino', 10.4],
  ],
  age: [
    ['Children under 10', 1.4],
    ['Ages 10 - 64', 80.3],
    ['Elderly over 65', 7.2],
  ],
};

const demographicItemGen = (demographicData: any[]) => {
  return demographicData.map((el, index) => {
    return (
      <div key={index} className={styles.demographicItem}>
        <span>{ el[0] }</span>
        <span>{`${el[1]}%`}</span>
      </div>
    );
  });
};

const Demographics = () => {
  const demographicSections = [
    {
      title: `Racial Ethnographic`,
      content: demographicItemGen(demographicsData.racial),
      expanded: false,
      id: '123',
      headingLevel: 'h4',
    },
    {
      title: `Age`,
      content: demographicItemGen(demographicsData.age),
      expanded: false,
      id: 'abc',
      headingLevel: 'h4',
    },
  ];

  return (
    <div className={styles.demographicsContainer}>
      <div>Tract demographics</div>
      <Accordion items={demographicSections} multiselectable={true}/>
    </div>
  );
};

export default Demographics;
