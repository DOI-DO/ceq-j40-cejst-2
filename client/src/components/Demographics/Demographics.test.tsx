import React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import Demographics from './Demographics';

describe('rendering of Demographics Component', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <Demographics />
      </LocalizedComponent>,
  );
  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
