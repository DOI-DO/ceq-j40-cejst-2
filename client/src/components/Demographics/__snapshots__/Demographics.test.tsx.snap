// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of Demographics Component checks if component renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      Tract demographics
    </div>
    <div
      aria-multiselectable="true"
      class="usa-accordion"
      data-testid="accordion"
    >
      <h4
        class="usa-accordion__heading"
      >
        <button
          aria-controls="123"
          aria-expanded="false"
          class="usa-accordion__button"
          data-testid="accordionButton_123"
          type="button"
        >
          Racial Ethnographic
        </button>
      </h4>
      <div
        class="usa-accordion__content usa-prose"
        data-testid="accordionItem_123"
        hidden=""
        id="123"
      >
        <div>
          <span>
            White
          </span>
          <span>
            61.4%
          </span>
        </div>
        <div>
          <span>
            Black or African Americon
          </span>
          <span>
            10.3%
          </span>
        </div>
        <div>
          <span>
            American Indian and Alaska Native
          </span>
          <span>
            10.3%
          </span>
        </div>
        <div>
          <span>
            American Indian and Alaska Native
          </span>
          <span>
            0.3%
          </span>
        </div>
        <div>
          <span>
            Asian
          </span>
          <span>
            7.2%
          </span>
        </div>
        <div>
          <span>
            Native Hawiian or Pacific Islander
          </span>
          <span>
            4.2%
          </span>
        </div>
        <div>
          <span>
            Other
          </span>
          <span>
            2.5%
          </span>
        </div>
        <div>
          <span>
            Two or more races
          </span>
          <span>
            6.7%
          </span>
        </div>
        <div>
          <span>
            Hispanic or Latino
          </span>
          <span>
            10.4%
          </span>
        </div>
      </div>
      <h4
        class="usa-accordion__heading"
      >
        <button
          aria-controls="abc"
          aria-expanded="false"
          class="usa-accordion__button"
          data-testid="accordionButton_abc"
          type="button"
        >
          Age
        </button>
      </h4>
      <div
        class="usa-accordion__content usa-prose"
        data-testid="accordionItem_abc"
        hidden=""
        id="abc"
      >
        <div>
          <span>
            Children under 10
          </span>
          <span>
            1.4%
          </span>
        </div>
        <div>
          <span>
            Ages 10 - 64
          </span>
          <span>
            80.3%
          </span>
        </div>
        <div>
          <span>
            Elderly over 65
          </span>
          <span>
            7.2%
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
